src/movableCoord.js

```diff
@@ -183,6 +183,10 @@ eg.module("movableCoord", ["jQuery", eg, window, "Hammer"], function($, ns, glob
 						this._subOptions = options;
 						this._status.curHammer = hammer;
 						this._panstart(e);
+					} else if (e.isFinal && e.direction !== MC.DIRECTION_NONE &&
+						!(e.direction & this._subOptions.direction)) {
+						// Although direction of movement is different from options.direction, MC should fire 'release' event.
+						this._panend(e);
```

Comment:
@sculove OK, I also think that _panend do much~
I'm happy to help you find the potential bug related with `moveDistance`.